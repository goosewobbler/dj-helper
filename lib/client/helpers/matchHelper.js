"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fp_1 = require("lodash/fp");
var normalise = function (value) { return fp_1.deburr(value.toLowerCase()); };
var getNeedleSubstring = function (needle, haystack) {
    var normalisedNeedle = normalise(needle);
    var normalisedHaystack = normalise(haystack);
    var indexOfFirstNeedleCharacter = normalisedHaystack.indexOf(normalisedNeedle[0]);
    var length = 0;
    if (indexOfFirstNeedleCharacter > -1) {
        length = 1;
        while (length < normalisedNeedle.length &&
            normalisedHaystack[indexOfFirstNeedleCharacter + length] === normalisedNeedle[length]) {
            length += 1;
        }
    }
    return { indexOfFirstNeedleCharacter: indexOfFirstNeedleCharacter, length: length };
};
var getMatchesRecursive = function (needle, haystack) {
    var _a = getNeedleSubstring(needle, haystack), indexOfFirstNeedleCharacter = _a.indexOfFirstNeedleCharacter, length = _a.length;
    if (length === 0) {
        return [haystack];
    }
    var start = haystack.substr(0, indexOfFirstNeedleCharacter);
    var matched = haystack.substr(indexOfFirstNeedleCharacter, length);
    var end = haystack.substr(indexOfFirstNeedleCharacter + length);
    return [start, { matched: matched }].concat(getMatchesRecursive(needle.substr(length), end)).filter(Boolean);
};
var getMatches = function (needle, haystack) {
    var trimmedNeedle = fp_1.trim(needle);
    var matches = getMatchesRecursive(trimmedNeedle, haystack);
    var matchedParts = matches.filter(function (match) { return typeof match === 'object'; });
    var combinedMatchedParts = matchedParts.reduce(function (acc, match) { return acc + match.matched; }, '');
    if (combinedMatchedParts.length !== trimmedNeedle.length) {
        return [haystack];
    }
    return matches;
};
var getMatchesWithAlternatives = function (needle, haystack, alternatives) {
    var matches = getMatches(needle, haystack);
    if (fp_1.some(fp_1.isObject)(matches)) {
        return matches;
    }
    var alternativeMatches = alternatives.map(function (alternative) { return getMatches(needle, alternative); });
    return fp_1.find(fp_1.some(fp_1.isObject), alternativeMatches);
};
exports.default = getMatchesWithAlternatives;
//# sourceMappingURL=matchHelper.js.map