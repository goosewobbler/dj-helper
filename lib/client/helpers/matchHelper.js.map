{"version":3,"file":"matchHelper.js","sourceRoot":"","sources":["../../../src/client/helpers/matchHelper.ts"],"names":[],"mappings":";;AAAA,gCAA+D;AAE/D,IAAM,SAAS,GAAG,UAAC,KAAa,IAAa,OAAA,WAAM,CAAC,KAAK,CAAC,WAAW,EAAE,CAAC,EAA3B,CAA2B,CAAC;AAEzE,IAAM,kBAAkB,GAAG,UAAC,MAAc,EAAE,QAAgB;IAC1D,IAAM,gBAAgB,GAAG,SAAS,CAAC,MAAM,CAAC,CAAC;IAC3C,IAAM,kBAAkB,GAAG,SAAS,CAAC,QAAQ,CAAC,CAAC;IAC/C,IAAM,2BAA2B,GAAG,kBAAkB,CAAC,OAAO,CAAC,gBAAgB,CAAC,CAAC,CAAC,CAAC,CAAC;IAEpF,IAAI,MAAM,GAAG,CAAC,CAAC;IAEf,IAAI,2BAA2B,GAAG,CAAC,CAAC,EAAE;QACpC,MAAM,GAAG,CAAC,CAAC;QACX,OACE,MAAM,GAAG,gBAAgB,CAAC,MAAM;YAChC,kBAAkB,CAAC,2BAA2B,GAAG,MAAM,CAAC,KAAK,gBAAgB,CAAC,MAAM,CAAC,EACrF;YACA,MAAM,IAAI,CAAC,CAAC;SACb;KACF;IAED,OAAO,EAAE,2BAA2B,6BAAA,EAAE,MAAM,QAAA,EAAE,CAAC;AACjD,CAAC,CAAC;AAEF,IAAM,mBAAmB,GAAG,UAAC,MAAc,EAAE,QAAgB;IACrD,IAAA,yCAA8E,EAA5E,4DAA2B,EAAE,kBAA+C,CAAC;IAErF,IAAI,MAAM,KAAK,CAAC,EAAE;QAChB,OAAO,CAAC,QAAQ,CAAC,CAAC;KACnB;IAED,IAAM,KAAK,GAAG,QAAQ,CAAC,MAAM,CAAC,CAAC,EAAE,2BAA2B,CAAC,CAAC;IAC9D,IAAM,OAAO,GAAG,QAAQ,CAAC,MAAM,CAAC,2BAA2B,EAAE,MAAM,CAAC,CAAC;IACrE,IAAM,GAAG,GAAG,QAAQ,CAAC,MAAM,CAAC,2BAA2B,GAAG,MAAM,CAAC,CAAC;IAElE,OAAO,CAAC,KAAK,EAAE,EAAE,OAAO,SAAA,EAAE,CAAC,CAAC,MAAM,CAAC,mBAAmB,CAAC,MAAM,CAAC,MAAM,CAAC,MAAM,CAAC,EAAE,GAAG,CAAC,CAAC,CAAC,MAAM,CAAC,OAAO,CAAC,CAAC;AACtG,CAAC,CAAC;AAEF,IAAM,UAAU,GAAG,UAAC,MAAc,EAAE,QAAgB;IAClD,IAAM,aAAa,GAAG,SAAI,CAAC,MAAM,CAAC,CAAC;IACnC,IAAM,OAAO,GAAG,mBAAmB,CAAC,aAAa,EAAE,QAAQ,CAAC,CAAC;IAC7D,IAAM,YAAY,GAAG,OAAO,CAAC,MAAM,CAAC,UAAA,KAAK,IAAI,OAAA,OAAO,KAAK,KAAK,QAAQ,EAAzB,CAAyB,CAA+B,CAAC;IACtG,IAAM,oBAAoB,GAAG,YAAY,CAAC,MAAM,CAAC,UAAC,GAAG,EAAE,KAAK,IAAK,OAAA,GAAG,GAAG,KAAK,CAAC,OAAO,EAAnB,CAAmB,EAAE,EAAE,CAAC,CAAC;IAE1F,IAAI,oBAAoB,CAAC,MAAM,KAAK,aAAa,CAAC,MAAM,EAAE;QACxD,OAAO,CAAC,QAAQ,CAAC,CAAC;KACnB;IAED,OAAO,OAAO,CAAC;AACjB,CAAC,CAAC;AAEF,IAAM,0BAA0B,GAAG,UAAC,MAAc,EAAE,QAAgB,EAAE,YAAsB;IAC1F,IAAM,OAAO,GAAG,UAAU,CAAC,MAAM,EAAE,QAAQ,CAAC,CAAC;IAE7C,IAAI,SAAI,CAAC,aAAQ,CAAC,CAAC,OAAO,CAAC,EAAE;QAC3B,OAAO,OAAO,CAAC;KAChB;IAED,IAAM,kBAAkB,GAAG,YAAY,CAAC,GAAG,CAAC,UAAA,WAAW,IAAI,OAAA,UAAU,CAAC,MAAM,EAAE,WAAW,CAAC,EAA/B,CAA+B,CAAC,CAAC;IAC5F,OAAO,SAAI,CAAC,SAAI,CAAC,aAAQ,CAAC,EAAE,kBAAkB,CAAC,CAAC;AAClD,CAAC,CAAC;AAEF,kBAAe,0BAA0B,CAAC"}