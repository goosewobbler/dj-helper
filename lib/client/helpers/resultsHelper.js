"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var fp_1 = require("lodash/fp");
var React = require("react");
var matchHelper_1 = require("./matchHelper");
var numberOfMatchedRegions = function (item) {
    return item.matches.reduce(function (acc, match) { return acc + (fp_1.isObject(match) ? 1 : 0); }, 0);
};
var itemStartsWithSignificantMatch = function (item) {
    return !(fp_1.isObject(item.matches[0]) && item.matches[0].matched.length > 1);
};
var alphabeticalSort = fp_1.sortBy(fp_1.get('displayName'));
var numberOfRegionsSort = fp_1.sortBy(numberOfMatchedRegions);
var startsWithSignificantMatchSort = fp_1.sortBy(itemStartsWithSignificantMatch);
var sortItems = function (list) { return startsWithSignificantMatchSort(numberOfRegionsSort(alphabeticalSort(list))); };
var createItemWithMatches = function (item, searchValue) {
    return fp_1.assign(item, {
        matches: matchHelper_1.default(searchValue, item.displayName, item.alternatives),
    });
};
var itemMatchesSearchValue = function (item) { return fp_1.some(fp_1.isObject)(item.matches); };
var findOrderedSearchResults = function (items, searchValue) {
    var itemsWithMatches = items.map(function (item) { return createItemWithMatches(item, searchValue); });
    var results = sortItems(itemsWithMatches.filter(itemMatchesSearchValue));
    return results.map(function (item) {
        return fp_1.assign(item, {
            highlighted: item.matches.map(function (match, index) {
                if (match.matched) {
                    return React.createElement("mark", { key: "match-" + index }, match.matched.replace(/ /g, '-'));
                }
                return match.replace(/ /g, '-');
            }),
        });
    });
};
exports.default = findOrderedSearchResults;
//# sourceMappingURL=resultsHelper.js.map