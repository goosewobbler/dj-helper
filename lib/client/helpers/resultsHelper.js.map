{"version":3,"file":"resultsHelper.js","sourceRoot":"","sources":["../../../src/client/helpers/resultsHelper.tsx"],"names":[],"mappings":";;AAAA,gCAAgE;AAChE,6BAA+B;AAE/B,6CAAuD;AAEvD,IAAM,sBAAsB,GAAG,UAAC,IAAS;IACvC,OAAA,IAAI,CAAC,OAAO,CAAC,MAAM,CAAC,UAAC,GAAQ,EAAE,KAAU,IAAK,OAAA,GAAG,GAAG,CAAC,aAAQ,CAAC,KAAK,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,CAAC,EAA/B,CAA+B,EAAE,CAAC,CAAC;AAAjF,CAAiF,CAAC;AAEpF,IAAM,8BAA8B,GAAG,UAAC,IAAS;IAC/C,OAAA,CAAC,CAAC,aAAQ,CAAC,IAAI,CAAC,OAAO,CAAC,CAAC,CAAC,CAAC,IAAI,IAAI,CAAC,OAAO,CAAC,CAAC,CAAC,CAAC,OAAO,CAAC,MAAM,GAAG,CAAC,CAAC;AAAlE,CAAkE,CAAC;AAErE,IAAM,gBAAgB,GAAG,WAAM,CAAC,QAAG,CAAC,aAAa,CAAC,CAAC,CAAC;AAEpD,IAAM,mBAAmB,GAAG,WAAM,CAAC,sBAAsB,CAAC,CAAC;AAE3D,IAAM,8BAA8B,GAAG,WAAM,CAAC,8BAA8B,CAAC,CAAC;AAE9E,IAAM,SAAS,GAAG,UAAC,IAAW,IAAK,OAAA,8BAA8B,CAAC,mBAAmB,CAAC,gBAAgB,CAAC,IAAI,CAAC,CAAC,CAAC,EAA3E,CAA2E,CAAC;AAE/G,IAAM,qBAAqB,GAAG,UAAC,IAAS,EAAE,WAAmB;IAC3D,OAAA,WAAM,CAAC,IAAI,EAAE;QACX,OAAO,EAAE,qBAA0B,CAAC,WAAW,EAAE,IAAI,CAAC,WAAW,EAAE,IAAI,CAAC,YAAY,CAAC;KACtF,CAAC;AAFF,CAEE,CAAC;AAEL,IAAM,sBAAsB,GAAG,UAAC,IAAS,IAAK,OAAA,SAAI,CAAC,aAAQ,CAAC,CAAC,IAAI,CAAC,OAAO,CAAC,EAA5B,CAA4B,CAAC;AAE3E,IAAM,wBAAwB,GAAG,UAAC,KAAY,EAAE,WAAmB;IACjE,IAAM,gBAAgB,GAAG,KAAK,CAAC,GAAG,CAAC,UAAC,IAAS,IAAK,OAAA,qBAAqB,CAAC,IAAI,EAAE,WAAW,CAAC,EAAxC,CAAwC,CAAC,CAAC;IAE5F,IAAM,OAAO,GAAG,SAAS,CAAC,gBAAgB,CAAC,MAAM,CAAC,sBAAsB,CAAC,CAAC,CAAC;IAE3E,OAAO,OAAO,CAAC,GAAG,CAAC,UAAC,IAAS;QAC3B,OAAA,WAAM,CAAC,IAAI,EAAE;YACX,WAAW,EAAE,IAAI,CAAC,OAAO,CAAC,GAAG,CAAC,UAAC,KAAU,EAAE,KAAa;gBACtD,IAAI,KAAK,CAAC,OAAO,EAAE;oBACjB,OAAO,8BAAM,GAAG,EAAE,WAAS,KAAO,IAAG,KAAK,CAAC,OAAO,CAAC,OAAO,CAAC,IAAI,EAAE,GAAG,CAAC,CAAQ,CAAC;iBAC/E;gBAED,OAAO,KAAK,CAAC,OAAO,CAAC,IAAI,EAAE,GAAG,CAAC,CAAC;YAClC,CAAC,CAAC;SACH,CAAC;IARF,CAQE,CACH,CAAC;AACJ,CAAC,CAAC;AAEF,kBAAe,wBAAwB,CAAC"}