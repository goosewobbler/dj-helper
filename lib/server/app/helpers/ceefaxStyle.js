"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ceefaxStyle = "<style>\n@font-face {\n  font-family: 'bbcmode7';\n  src: url('http://localhost:3333/bbcmode7.woff') format('woff'),\n       url('http://localhost:3333/bbcmode7.ttf') format('truetype');\n  font-weight: normal;\n  font-style: normal;\n}\nbody {\n  font-family: bbcmode7 !important;\n  background-color: black !important;\n  filter: invert(100%);\n}\n.sp-c-global-header .gel-wrap {\n  color: #18ca0f !important;\n  filter: invert(100%);\n}\n.sp-c-sport-navigation__link {\n  color: #18ca0f !important;\n  filter: invert(100%);\n}\n</style>";
exports.default = ceefaxStyle;
//# sourceMappingURL=ceefaxStyle.js.map