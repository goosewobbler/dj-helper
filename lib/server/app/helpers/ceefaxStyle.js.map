{"version":3,"file":"ceefaxStyle.js","sourceRoot":"","sources":["../../../../src/server/app/helpers/ceefaxStyle.ts"],"names":[],"mappings":";;AAAA,IAAM,WAAW,GAAG,6hBAqBX,CAAC;AAEV,kBAAe,WAAW,CAAC"}