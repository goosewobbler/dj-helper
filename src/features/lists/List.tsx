import React, { ReactElement, BaseSyntheticEvent, KeyboardEvent } from 'react';

type ListProps = {
  id: number;
  title: string;
  editing?: boolean;
  onTitleChange: (id: number, titleValue: string) => void;
  onClickEdit: (id: number) => void;
  onClickDelete: (id: number) => void;
  onEditingComplete: (id: number, titleValue: string) => void;
  onEditingCancelled: (id: number) => void;
};

const KEY_ENTER = 'Enter';
const KEY_ESCAPE = 'Escape';

export function List({
  id,
  title,
  editing,
  onTitleChange,
  onEditingCancelled,
  onClickEdit,
  onClickDelete,
  onEditingComplete,
}: ListProps): ReactElement {
  const isValid = () => {
    return !!title;
  };

  const handleTitleChange = (event: BaseSyntheticEvent): void => {
    onTitleChange(id, (event.target as HTMLInputElement).value);
  };

  const handleKeyDown = (event: KeyboardEvent, listId: number): void => {
    if (event.key === KEY_ENTER && isValid()) {
      onEditingComplete(listId, (event.target as HTMLInputElement).value);
    } else if (event.key === KEY_ESCAPE) {
      onEditingCancelled(listId);
    }
  };

  const validityRing = `ring-${isValid() ? 'green' : 'red'}-300`;

  return (
    <li id={`list-${id}`} className="list" role="listitem">
      <span className="title" data-testid="title">
        {editing ? (
          <>
            <label htmlFor="list-title">List Title</label>
            <input
              id="list-title"
              className={`ring-4 ring-opacity-30 ${validityRing}`}
              type="text"
              onChange={(event): void => handleTitleChange(event)}
              onKeyDown={(event): void => handleKeyDown(event, id)}
              placeholder="List Title"
              value={title}
            />
          </>
        ) : (
          title
        )}
      </span>
      {!editing && (
        <span className="action-btns">
          <button className="p-1 border" type="button" onClick={() => onClickEdit(id)} role="button">
            Edit
          </button>
          <button className="p-1 border" type="button" onClick={() => onClickDelete(id)} role="button">
            Delete
          </button>
        </span>
      )}
    </li>
  );
}
