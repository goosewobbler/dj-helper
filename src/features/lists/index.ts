import type { AppState, AppStore, List, Track } from '../../common/types.js';

const initialListTitle = 'New List';

function reorderTracks(tracks: List['tracks'], trackIdToMove: Track['id'], newIndex: number) {
  const indexOfTrackIdToMove = tracks.indexOf(trackIdToMove);
  if (indexOfTrackIdToMove === -1) {
    return tracks;
  }

  // clone array and remove track
  const otherTracks = Array.from(tracks);
  otherTracks.splice(indexOfTrackIdToMove, 1);

  // insert track at the desired index
  const tracksBefore = otherTracks.slice(0, newIndex);
  const tracksAfter = otherTracks.slice(newIndex);
  return tracksBefore.concat([trackIdToMove]).concat(tracksAfter);
}

export const initialState: List[] = [
  {
    id: 0,
    title: 'Look a set list',
    tracks: [],
    editing: false,
    active: true,
  },
  {
    id: 1,
    title: 'Wow such list',
    tracks: [],
    editing: false,
    active: false,
  },
];

export const handlers = (store: AppStore) => ({
  'LIST:CREATE': () => {
    store.setState((state) => {
      const newList: List = {
        id: state.lists.length,
        title: initialListTitle,
        tracks: [],
        editing: true,
        active: true,
      };
      return { lists: [...state.lists.map((list) => ({ ...list, active: false })), newList] };
    });
  },
  'LIST:UPDATE_TITLE': ({ id, title }: { id: number; title: string }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) => (list.id === id ? { ...list, title } : list)),
    })),
  'LIST:DELETE': ({ id }: { id: number }) =>
    store.setState((state) => ({
      lists: state.lists.filter((list) => list.id !== id).map((list, index) => ({ ...list, id: index })),
    })),
  'LIST:EDIT': ({ id }: { id: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) => (list.id === id ? { ...list, editing: true, oldTitle: list.title } : list)),
    })),
  'LIST:EDIT_COMPLETE': () =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.editing === true ? { ...list, editing: undefined, oldTitle: undefined } : list,
      ),
    })),
  'LIST:EDIT_REVERT': () =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.editing === true
          ? { ...list, editing: undefined, title: list.oldTitle ?? initialListTitle, oldTitle: undefined }
          : list,
      ),
    })),
  'LIST:TOGGLE_ACTIVE': ({ id }: { id: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.id === id ? { ...list, active: !list.active } : { ...list, active: false },
      ),
    })),
  'LIST:ADD_TRACK': ({ trackId }: { trackId: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) => (list.active ? { ...list, tracks: [...list.tracks, trackId] } : list)),
    })),
  'LIST:REMOVE_TRACK': ({ trackId }: { trackId: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.active ? { ...list, tracks: list.tracks.filter((track) => track !== trackId) } : list,
      ),
    })),
  'LIST:MOVE_TRACK': ({ trackId, newIndex }: { trackId: number; newIndex: number }) =>
    store.setState((state) => ({
      lists: state.lists.map((list) =>
        list.active ? { ...list, tracks: reorderTracks(list.tracks, trackId, newIndex) } : list,
      ),
    })),
});

export const selectLists = (state: Partial<AppState>) => state.lists as List[];

export const selectActiveList = (state: Partial<AppState>): List => state.lists?.find((list) => list.active) as List;

export const selectListById =
  (id: number) =>
  (state: Partial<AppState>): List =>
    state.lists?.find((list) => list.id === id) as List;

export const trackIsOnActiveList =
  ({ trackId }: { trackId: number }) =>
  (state: Partial<AppState>): boolean =>
    (state.lists?.find((list) => list.active) as List)?.tracks.some((track) => track === trackId);

export const getNextTrackOnList =
  ({ id, currentTrackId }: { id: number; currentTrackId: number }) =>
  (state: Partial<AppState>): Track['id'] | undefined => {
    const list = state.lists?.find((stateList) => stateList.id === id) as List;
    const listTracks = (state.lists?.find((stateList) => stateList.id === id) as List).tracks;
    if (!list) {
      return undefined;
    }
    const currentTrackIndex = list.tracks.findIndex((track) => track === currentTrackId);
    return listTracks[currentTrackIndex + 1];
  };

export const getPreviousTrackOnList =
  ({ id, currentTrackId }: { id: number; currentTrackId: number }) =>
  (state: Partial<AppState>): Track['id'] | undefined => {
    const list = state.lists?.find((stateList) => stateList.id === id) as List;
    const listTracks = (state.lists?.find((stateList) => stateList.id === id) as List).tracks;
    if (!list) {
      return undefined;
    }
    const currentTrackIndex = listTracks.findIndex((track) => track === currentTrackId);
    return listTracks[currentTrackIndex - 1];
  };
