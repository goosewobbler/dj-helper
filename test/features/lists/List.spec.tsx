import * as React from 'react';
import { render, screen, RenderResult, userEvent } from '../../helpers/integration';
import { List } from '../../../src/features/lists/List';

let list: RenderResult;

describe('List', () => {
  let onClickDeleteMock: jest.Mock;
  let onClickEditMock: jest.Mock;

  beforeEach(() => {
    onClickDeleteMock = jest.fn();
    onClickEditMock = jest.fn();
    list = render(<List id={546} />, {
      preloadedState: {
        lists: [{ id: 546, title: 'list one', tracks: [] }],
        embed: { isPlaying: false },
        tracks: [],
        browsers: [],
      },
    });
  });

  it('should render a List with the expected title', () => {
    expect(screen.getByTestId('title')).toHaveTextContent('list one');
  });

  it('should render the expected html', () => {
    expect(list.container).toMatchSnapshot();
  });

  describe('when the delete button is clicked', () => {
    beforeEach(() => {
      const deleteBtn = screen.getByRole('button', { name: /Delete/i });
      userEvent.click(deleteBtn);
    });

    it('should fire the delete click handler with the List ID', () => {
      expect(onClickDeleteMock).toHaveBeenCalledWith(546);
    });

    it('should render the expected html', () => {
      expect(list.container).toMatchSnapshot();
    });
  });

  describe('when the edit button is clicked', () => {
    beforeEach(() => {
      const editBtn = screen.getByRole('button', { name: /Edit/i });
      userEvent.click(editBtn);
    });

    it('should fire the edit click handler with the List ID', () => {
      expect(onClickEditMock).toHaveBeenCalledWith(546);
    });

    it('should render the expected html', () => {
      expect(list.container).toMatchSnapshot();
    });
  });
});
