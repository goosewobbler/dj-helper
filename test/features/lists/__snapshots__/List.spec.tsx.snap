// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render the expected html 1`] = `
<div>
  <div
    class="list"
  >
    <span
      class="title"
      data-testid="title"
    >
      test list
    </span>
    <span
      class="action-btns"
    >
      <button
        data-testid="edit"
        type="button"
      >
        Edit
      </button>
      <button
        data-testid="delete"
        type="button"
      >
        Delete
      </button>
    </span>
  </div>
</div>
`;

exports[`List when the delete button is clicked should render the expected html 1`] = `
<div>
  <div
    class="list"
  >
    <span
      class="title"
      data-testid="title"
    >
      test list
    </span>
    <span
      class="action-btns"
    >
      <button
        data-testid="edit"
        type="button"
      >
        Edit
      </button>
      <button
        data-testid="delete"
        type="button"
      >
        Delete
      </button>
    </span>
  </div>
</div>
`;
